components/script/dom/element.rs

```diff
@@ -856,6 +871,329 @@ impl Element {
         let result = media_list.evaluate(document.window().layout().device(), quirks_mode);
         result
     }
+
+    /// <https://drafts.csswg.org/cssom-view/#scroll-a-target-into-view>
+    fn scroll_into_view_with_options(
+        &self,
+        behavior: ScrollBehavior,
+        block: ScrollLogicalPosition,
+        inline: ScrollLogicalPosition,
+        container: Option<&Element>,
+        can_gc: CanGc,
+    ) {
+        let target_document = self.upcast::<Node>().owner_doc();
+
+        // Step 1: For each ancestor element or viewport that establishes a scrolling box,
+        // in order of innermost to outermost scrolling box
+        self.upcast::<Node>()
+            .inclusive_ancestors(ShadowIncluding::Yes)
+            .skip(1) // Skip self
+            .filter(|node| self.establishes_scroll_box(node, can_gc)) // Check if node establishes a scrolling box
```

Comment:
```suggestion
            .filter(|node| self.establishes_scroll_box(node, can_gc))
```
nit: the comment seems redundant